Makes the value of annotations optional

artman opened the pull request:
There are cases when annotation values aren't useful, such as marking a fields as deprecated. This PR adds support for annotations without values, so that one can write a struct like so:

```
struct MyStruct {
  1: i32 num1 ( deprecated )
}
```

phated commented:
Have Union & Exception tests locally (duplicates of struct tests due to same parsing) but waiting for #50 to land before pushing so I can rebase it in.

Last major tests to write are Service tests (which are currently missing `extends` because it hasn't been merged upstream yet).

phated commented:
Have Service tests locally now too.

phated commented:
See #55 for the updated version.

Commit:
Makes the value of annotations optional

There are cases when annotation values aren't useful, such as marking a fields as deprecated. This PR adds support for annotations without values, so that one can write a struct like so:

```
struct MyStruct {
  1: i32 num1 ( deprecated )
}
```
